Lib/ipaddress.py

```diff
@@ -975,6 +972,34 @@ def is_multicast(self):
         return (self.network_address.is_multicast and
                 self.broadcast_address.is_multicast)
 
+    def _containment_check(self, check, other):
```

Comment:
Instead of switching on `check`, how about having something like:
```python
@classmethod
def _is_subnet_of(cls, a, b):
   # Implementation here

def subnet_of(self, other):
   return self._is_subnet_of(self, other)

def supernet_of(self, other):
   return self._is_subnet_of(other, self)
```

Reply:
Thank you.  I made these changes, except I made is_subnet_of a staticmethod since it didn't use cls.